zap/src/main/java/org/parosproxy/paros/core/scanner/Alert.java

```diff
@@ -1087,6 +1089,59 @@ public Builder setTags(Map<String, String> tags) {
             return this;
         }
 
+        /**
+         * Adds an Alert tag with the given key (tag/name) to the existing collection for this
+         * Alert/Builder.
+         *
+         * @since 2.12.0
+         */
+        public Builder addTag(String tag) {
+            addTag(tag, "");
+            return this;
+        }
+
+        /**
+         * Adds an Alert tag with the given key (tag/name) and value to the existing collection for
+         * this Alert/Builder.
+         *
+         * @since 2.12.0
+         */
+        public Builder addTag(String tag, String value) {
```

Comment:
This method (and the others) could be synchronized to avoid potential race conditions. The `HashMap` could be replaced with a `ConcurrentHashMap`.

Reply:
In which case will the concurrent access happen? The builders should not be being shared.

Reply:
I don't have an exact case in mind but just being safe. Maybe builders accessed from scripts or multithreaded passive scan rules?

Reply:
They would still have their own builder in those cases.

Reply:
Damn, my bad, that makes sense. Thanks for the clarification.